xbmc/cores/VideoPlayer/VideoRenderers/windows/RendererDXVA.cpp

```diff
@@ -31,9 +31,25 @@ void CRendererDXVA::GetWeight(std::map<RenderMethod, int>& weights, const VideoP
 {
   unsigned weight = 0;
   const AVPixelFormat av_pixel_format = picture.videoBuffer->GetFormat();
+  const DXGI_FORMAT dxgi_format =
+      CRenderBufferImpl::GetDXGIFormat(av_pixel_format, GetDXGIFormat(picture));
+
+  const bool streamIsHDR = (picture.color_primaries == AVCOL_PRI_BT2020) &&
+                           (picture.color_transfer == AVCOL_TRC_SMPTE2084 ||
+                            picture.color_transfer == AVCOL_TRC_ARIB_STD_B67);
+  const auto settings = CServiceBroker::GetSettingsComponent()->GetSettings();
+  const auto setting = DX::Windowing()->SETTING_WINSYSTEM_IS_HDR_DISPLAY;
+  const bool systemUsesHDR =
+      (settings ? settings->GetBool(setting) && DX::Windowing()->IsHDRDisplay() : false) ||
+      DX::Windowing()->IsHDROutput();
 
   if (av_pixel_format == AV_PIX_FMT_D3D11VA_VLD)
   {
+    // Check if HDR10 passthrough is supported by DXVA video processor
+    if (streamIsHDR && systemUsesHDR &&
```

Comment:
Please add comment that the check is relevant also for HLG because the renderer has a workaround to use PQ passthrough when the processor doesn't support HLG input.